auth/presentation/src/main/java/com/pronoidsoftware/auth/presentation/login/LoginState.kt

```diff
@@ -0,0 +1,50 @@
+package com.pronoidsoftware.auth.presentation.login
+
+import androidx.compose.foundation.text.input.TextFieldState
+import androidx.compose.runtime.derivedStateOf
+
+data class LoginState(
+    val email: TextFieldState = TextFieldState(),
+    val isEmailValid: Boolean = false,
+    val password: TextFieldState = TextFieldState(),
+    val isPasswordVisible: Boolean = false,
+    val isLoggingIn: Boolean = false,
+) {
+    val canLogin = derivedStateOf {
+        !isLoggingIn && isEmailValid && password.text.isNotBlank()
+    }
```

Comment:
I think you could use the delegate here, since otherwise you get a `State<Boolean>` don't you?

Reply:
ah yes, that cleans things up a bit, thanks!